### PR TITLE
Report operator input distribution

### PR DESCRIPTION
**Pull request by sopel39**
This makes it easier to track skewness/CPU utilization issues

### CODE REVIEW

**Comment by sopel39** [comment]
fyi: @dain @martint

File: core/trino-main/src/main/java/io/trino/operator/WorkProcessorPipelineSourceOperator.java [hunk 1]
```diff
@@ -332,7 +332,7 @@ private void updatePeakMemoryReservations(int operatorIndex)
 
                         succinctBytes(context.inputDataSize.get()),
                         context.inputPositions.get(),
-                        context.inputPositions.get() * context.inputPositions.get(),
+                        (double) context.inputPositions.get() * context.inputPositions.get(),
```
Comment (losipiuk) [inline core/trino-main/src/main/java/io/trino/operator/WorkProcessorPipelineSourceOperator.java:336]: nit:
commit message should be sth like `Prevent integer overflow when computing statistics`

File: core/trino-main/src/main/java/io/trino/operator/Distribution.java [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+    private final double max;
+    private final double p01;
+    private final double p05;
+    private final double p10;
+    private final double p25;
+    private final double p50;
+    private final double p75;
+    private final double p90;
+    private final double p95;
+    private final double p99;
```
Comment (losipiuk) [inline core/trino-main/src/main/java/io/trino/operator/Distribution.java:46]: Why materialize specific percentile values instead keeping TDigest?

File: core/trino-main/src/main/java/io/trino/operator/Distribution.java [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+    public static Distribution withCount(long count)
```
Comment (losipiuk) [inline core/trino-main/src/main/java/io/trino/operator/Distribution.java:48]: `with...` suggests it is an instance method which adds `count` information to the instance. Maybe name it `ofCount`

**Review by losipiuk — APPROVED** [review]

**Comment by losipiuk** [comment]
Fix CI - related

**Comment by sopel39** [comment]
> Fix CI - related

Fixed, concurrency issue

**Comment by sopel39** [comment]
Closing in favor of https://github.com/trinodb/trino/pull/10124

**Pull request closed without merging**